ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/CCodeGenerator.java

```diff
@@ -107,7 +105,30 @@ public void generateRequest(final Ds3Request ds3Request) throws IOException, Tem
         }
     }
 
+    public void generateType(final Map.Entry<String, Ds3Type> typeEntry) throws IOException {
```

Comment:
It does not appear that the map entry is needed here.  The code would be cleaner if either the map value (type name) or the map key (Ds3Type) was passed in instead of both.